django_project/realtime/tasks/flood.py

```diff
@@ -33,9 +48,63 @@
 LOGGER = logging.getLogger(LOGGER_NAME)
 
 
+@app.task(queue='inasafe-django')
+def check_processing_task():
+    """Checking flood processing task."""
+    # Checking analysis task
+    for flood in Flood.objects.exclude(
+            analysis_task_id__isnull=True).exclude(
+            analysis_task_id__exact='').filter(
+            analysis_task_status__iexact='PENDING'):
+        analysis_task_id = flood.analysis_task_id
+        result = AsyncResult(id=analysis_task_id, app=headless_app)
+        if result.state == 'SUCCESS':
+            task_state = 'FAILURE'
+            if result.result['status'] != 0:
+                LOGGER.error(result.result['message'])
+            else:
+                try:
+                    flood.impact_file_path = result.result['output'][
+                        'analysis_summary']
+                    task_state = 'SUCCESS'
+                except BaseException as e:
+                    LOGGER.exception(e)
+                flood.analysis_task_status = task_state
+                flood.save()
+    # Checking report generation task
+    for flood in Flood.objects.exclude(
+            report_task_id__isnull=True).exclude(
+            report_task_id__exact='').filter(
+            report_task_status__iexact='PENDING'):
+        report_task_id = flood.report_task_id
+        result = AsyncResult(id=report_task_id, app=headless_app)
+        if result.state == 'SUCCESS':
+            task_state = 'FAILURE'
+            try:
+                if result.result['status'] != 0:
+                    LOGGER.error(result.result['message'])
+                else:
+                    report_path = result.result[
+                        'output']['pdf_product_tag']['realtime-flood-en']
+                    # Create flood report object
+                    # Set the language manually first
+                    flood_report = FloodReport(flood=flood, language='en')
+                    with open(report_path, 'rb') as report_file:
+                        flood_report.impact_map.save(
+                            flood_report.impact_map_filename,
+                            File(report_file),
+                            save=True)
+                    flood_report.save()
```

Comment:
Not needed. You already put `save=True) above. CMIIW

Reply:
You are right.